### PR TITLE
ci: add CI & end-to-end tests and prepare initial goreleaser configuration

### PR DESCRIPTION
**Pull request by sheldonhull**
- Add integration test pipeline.
- Add e2e test pipeline.
- Both pipelines are split into seperate azure pipeline runs.
- Add goreleaser configs, this will be finalized in another pull request.
- Improve testing docs thanks to contribution from @andrii-zakurenyi.
- Add changie and changelog generation process.
- Add remaining workflow automation such as scans and more.

### CODE REVIEW

File: pipelines/templates/task-go-install.yml [hunk 1]
```diff
@@ -8,30 +15,42 @@ steps:
...
   # Prepend path: https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md
   - pwsh: |
-      $ENV:PATH="${ENV:HOME}/AppData/Local/aquaproj-aqua/bin;${ENV:PATH}" # ensure aqua managed binaries can be found
+      # $ENV:PATH = $ENV:AQUA_ROOT_DIR, 'bin', $ENV:PATH -join [IO.Path]::PathSeparator
+
+      Write-Host "Go binary used: $((Get-Command go -ErrorAction SilentlyContinue).Path)"
+      Write-Host "Go Version Used: $(go version)"
       New-Item -Path "$(GOBIN)" -ItemType Directory -Force -ErrorAction SilentlyContinue
-      &go env
+      $ENV:GOROOT= "$(&aqua which go)" env GOROOT
```
Comment (sheldonhull) [inline pipelines/templates/task-go-install.yml]:
```suggestion
      $ENV:GOROOT= &"$(&aqua which go)" env GOROOT
```

File: pipelines/templates/task-go-install.yml [hunk 2]
```diff
@@ -8,30 +15,43 @@ steps:
...
   - pwsh: |
-      $ENV:PATH="${ENV:HOME}/AppData/Local/aquaproj-aqua/bin;${ENV:PATH}" # ensure aqua managed binaries can be found
+      $ENV:PATH = $ENV:AQUA_ROOT_DIR, 'bin', $ENV:PATH -join [IO.Path]::PathSeparator
+
+      Write-Host "Go binary used: $((Get-Command go -ErrorAction SilentlyContinue).Path)"
+      Write-Host "Go Version Used: $(go version)"
       New-Item -Path "$(GOBIN)" -ItemType Directory -Force -ErrorAction SilentlyContinue
-      &go env
+
+      $ENV:GOROOT = Invoke-Expression ('{0} env GOROOT' -f "$(&aqua which go)")
```
Comment (sheldonhull) [inline pipelines/templates/task-go-install.yml]:
```suggestion
      $ENV:GOROOT = Invoke-Expression ('{0} env GOROOT' -f "$(&aqua which go | Resolve-Path -Relative )")
```

File: pipelines/templates/task-aqua-install.yml [hunk 3]
```diff
@@ -9,15 +9,35 @@ parameters:
...
-      curl -sSfL https://raw.githubusercontent.com/aquaproj/aqua-installer/v1.1.2/aqua-installer | bash -s -- -i "$(Agent.ToolsDirectory)/aqua"
-      export AQUA_PACKAGE_BIN="${AQUA_ROOT_DIR:-${XDG_DATA_HOME:-$HOME/.local/share}/aquaproj-aqua}/bin"
-      echo "##vso[task.prependpath]${AQUA_PACKAGE_BIN}"
-      echo "Added ${AQUA_PACKAGE_BIN} to PATH"
+  - pwsh: |
+      $binarydirectory = '${{ parameters.AQUA_BINARY_FULL_INSTALL_PATH }}' | Split-Path -Parent
+      New-Item -ItemType Directory -Path $binarydirectory -ErrorAction SilentlyContinue && Write-Host "✅ Created the directory to house the aqua binary: [$binarydirectory]"
+      New-Item -ItemType Directory -Path '${{ parameters.AQUA_ROOT_DIR }}' -Force -ErrorAction SilentlyContinue && Write-Host "✅ Created the directory to house the aqua binary: [${{ parameters.AQUA_ROOT_DIR }}]"
+      # go run github.com/aquaproj/aqua-installer@latest -o "${{ parameters.AQUA_BINARY_FULL_INSTALL_PATH }}"
+      &curl -sSfL https://raw.githubusercontent.com/aquaproj/aqua-installer/v1.1.2/aqua-installer | bash -s -- -i "${{ parameters.AQUA_BINARY_FULL_INSTALL_PATH }}"
```
Comment (sheldonhull) [inline pipelines/templates/task-aqua-install.yml]:
```suggestion
      go run github.com/aquaproj/aqua-installer@latest -o "${{ parameters.AQUA_BINARY_FULL_INSTALL_PATH }}"
      # &curl -sSfL https://raw.githubusercontent.com/aquaproj/aqua-installer/v1.1.2/aqua-installer | bash -s -- -i "${{ parameters.AQUA_BINARY_FULL_INSTALL_PATH }}"
```

File: pipelines/templates/task-aqua-install.yml [hunk 4]
```diff
@@ -9,15 +9,35 @@ parameters:
...
+      Write-Host "##vso[task.prependpath]${{ parameters.AQUA_ROOT_DIR }}/bin"
+      Write-Host "##vso[task.prependpath]$binarydirectory"
     displayName: install-aqua
-    condition: ne(variables['Agent.OS'], 'Windows_NT')
-  - bash: |
-      $(Agent.ToolsDirectory)/aqua install ${{ parameters.aquaInstallMethod }} ${{ parameters.aquaInstallTags }}
+    failOnStderr: false
+    env:
+      AQUA_GITHUB_TOKEN: ${{ parameters.AQUA_GITHUB_TOKEN }}
+      GOROOT: ${{ parameters.GOROOT }}
```
Comment (sheldonhull) [inline pipelines/templates/task-aqua-install.yml]:
```suggestion
      # GOROOT: ${{ parameters.GOROOT }}
```

Shouldn't need on install stage

**Review by andrii-zakurenyi — APPROVED** [review]

**Review by tdillenbeck — APPROVED** [review]
Great work!

**Pull request merged**